src/app/route/Checkout/Checkout.container.js

```diff
@@ -202,7 +203,10 @@ export class CheckoutContainer extends PureComponent {
         // Handle going back from billing to shipping
         if (/shipping/.test(urlStep) && /billing/.test(prevUrlStep)) {
             // eslint-disable-next-line react/no-did-update-set-state
-            this.setState({ checkoutStep: SHIPPING_STEP });
+            this.setState({
+                checkoutStep: SHIPPING_STEP,
+                isGuestEmailSaved: false
```

Comment:
It is impossible to change email if we go back from billing to shipping step

Reply:
I think we could resolve it as a separate bug. Here: https://github.com/scandipwa/base-theme/issues/1574

Reply:
Oops, sorry seems like this actually solves that. Closing with reference here :D